cmd/config/internal/commands/annotate_test.go

```diff
@@ -0,0 +1,469 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package commands
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+)
+
+func TestAnnotateCommand(t *testing.T) {
+	var tests = []struct {
+		name     string
+		args     []string
+		expected string
+	}{
+		{
+			name: "single value",
+			args: []string{"--kv", "a=b"},
+			expected: `kind: Deployment
+metadata:
```

Comment:
Nit: I think this might be more readable if you create variables for the yaml strings rather than have it inline in the test.

Reply:
Done